weave/scorers/trust_scorer.py

```diff
@@ -0,0 +1,376 @@
+                        scorer.score, **self._filter_inputs_for_scorer(scorer, inputs)
+                    ): scorer_name
+                    for scorer_name, scorer in self._loaded_scorers.items()
+                }
+                # Collect results as they complete.
+                for future in as_completed(future_to_scorer):
+                    scorer_name = future_to_scorer[future]
+                    try:
+                        results[scorer_name] = future.result()
+                    except Exception as e:
+                        raise WeaveTrustScorerError(
+                            f"Error calling {scorer_name}: {e}", errors=e
+                        )
+        else:
+            # Run scorers sequentially
+            for scorer_name, scorer in self._loaded_scorers.items():
+                try:
+                    results[scorer_name] = scorer.score(
+                        **self._filter_inputs_for_scorer(scorer, inputs)
+                    )
+                except Exception:
+                    pass
+
+        return results
+
+    def _score_with_logic(
+        self,
+        query: Optional[str],
+        context: Optional[Union[str, list[str]]],
+        output: str,
+    ) -> dict[str, Any]:
+        """Score with nuanced logic for trustworthiness."""
+        raw_results = self._score_all(output=output, context=context, query=query)
+
+        # Handle error case
+        if "error" in raw_results:
+            return raw_results["error"]
+
+        # Track issues by type
+        critical_issues = []
+        advisory_issues = []
+
+        # Check each scorer's results
+        for scorer_name, result in raw_results.items():
+            if not result.passed:
+                scorer_cls = type(self._loaded_scorers[scorer_name])
+                if scorer_cls in self._critical_scorers:
+                    critical_issues.append(scorer_name)
+                elif scorer_cls in self._advisory_scorers:
+                    advisory_issues.append(scorer_name)
+
+        # Determine trust level
+        passed = True
+        trust_level = "high_no-issues-found"
+        if critical_issues:
+            passed = False
+            trust_level = "low_critical-issues-found"
+        elif advisory_issues:
+            trust_level = "medium_advisory-issues-found"
+
+        # Extract scores where available
+        scores = {}
+        for name, result in raw_results.items():
+            if name == "WeaveToxicityScorerV1":
+                scores[name] = result.extras  # Toxicity returns category scores
+            elif hasattr(result, "extras") and "score" in result.extras:
+                scores[name] = result.extras["score"]
+
+        return {
+            "passed": passed,
+            "trust_level": trust_level,
+            "critical_issues": critical_issues,
+            "advisory_issues": advisory_issues,
+            "extras": {"raw_outputs": raw_results, "scores": scores},
+        }
```

Comment:
Why not just return the metadata dict that you use below?  Seems like you have to reconstruct it needlessly

Reply:
yeap, remove this